src/parser.rs

```diff
@@ -0,0 +1,169 @@
+use std::borrow::Cow;
+use std::str::{CharIndices, FromStr};
+
+use chrono::NaiveDate;
+use nom::bytes::complete::{is_not, tag, take};
+use nom::character::complete::{line_ending, one_of, space0};
+use nom::combinator::{iterator, map, map_opt, map_res, opt};
+use nom::sequence::{delimited, preceded, terminated};
+use nom::{IResult, Parser};
+use nom_locate::position;
+
+use crate::todotxt::{Located, Priority, Span, Tag, Todo};
+
+pub type Error<'a> = nom::error::Error<Input<'a>>;
+pub type Input<'a> = nom_locate::LocatedSpan<&'a str>;
+
+struct SplitWhitespace<'a> {
+    iter: CharIndices<'a>,
+    len: usize,
+}
+
+/// Parse a todo list from the provided `&str`.
+///
+pub fn from_str(value: &str) -> impl Iterator<Item = Todo> {
+    iterator(value.into(), delimited(eol, todo(), eol))
+}
+
+pub fn tags(at: Span, description: &str) -> impl Iterator<Item = Tag> {
+    let line = at.line();
+    let offset = at.start();
+
+    SplitWhitespace::new(description).filter_map(move |(start, end)| {
```

Comment:
This will (probably) get refactored to use nom like the rest of the parser.

Too lazy to do it right now and want the main branch to show all the hard work I put in this week.